pkg/operator/controller/controller.go

```diff
@@ -41,157 +30,3 @@ const (
-	log.FromContext(ctx).Info("Starting controller")
-	defer log.FromContext(ctx).Info("Stopping controller")
-
-	for {
-		select {
-		case <-time.After(s.reconcile(ctx)):
-		case <-ctx.Done():
-			return nil
-		}
-	}
-}
-
-func (s *Singleton) reconcile(ctx context.Context) time.Duration {
-	activeWorkers.WithLabelValues(s.name).Inc()
-	defer activeWorkers.WithLabelValues(s.name).Dec()
-
-	measureDuration := metrics.Measure(reconcileDuration.WithLabelValues(s.name))
-	res, err := s.Reconcile(ctx, singletonRequest)
-	measureDuration() // Observe the length of time between the function creation and now
-
-	switch {
-	case err != nil:
-		reconcileErrors.WithLabelValues(s.name).Inc()
-		reconcileTotal.WithLabelValues(s.name, labelError).Inc()
-		log.FromContext(ctx).Error(err, "Reconciler error")
-		return s.rateLimiter.When(singletonRequest)
-	case res.Requeue:
-		reconcileTotal.WithLabelValues(s.name, labelRequeue).Inc()
-		return s.rateLimiter.When(singletonRequest)
-	default:
-		s.rateLimiter.Forget(singletonRequest)
-		switch {
-		case res.RequeueAfter > 0:
-			reconcileTotal.WithLabelValues(s.name, labelRequeueAfter).Inc()
-			return res.RequeueAfter
-		default:
-			reconcileTotal.WithLabelValues(s.name, labelSuccess).Inc()
-			return time.Duration(0)
-		}
-	}
-}
-
-func (s *Singleton) NeedLeaderElection() bool {
-	return true
-}
-
-func init() {
-	mergeMetrics()
-}
-
-const (
-	labelError        = "error"
-	labelRequeueAfter = "requeue_after"
-	labelRequeue      = "requeue"
-	labelSuccess      = "success"
-)
-
-// Metrics below are copied metrics fired by controller-runtime in its /internal package. This is leveraged
-// so that we can fire to the same namespace as users expect other controller-runtime metrics to be fired
-// https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/internal/controller/metrics/metrics.go
-var (
-	reconcileTotal = prometheus.NewCounterVec(prometheus.CounterOpts{
-		Name: "controller_runtime_reconcile_total",
-		Help: "Total number of reconciliations per controller",
-	}, []string{"controller", "result"})
-	reconcileDuration = prometheus.NewHistogramVec(prometheus.HistogramOpts{
-		Name:    "controller_runtime_reconcile_time_seconds",
-		Help:    "Length of time per reconciliation per controller",
-		Buckets: metrics.DurationBuckets(),
-	}, []string{"controller"})
-	reconcileErrors = prometheus.NewCounterVec(prometheus.CounterOpts{
-		Name: "controller_runtime_reconcile_errors_total",
-		Help: "Total number of reconciliation errors per controller",
-	}, []string{"controller"})
-	workerCount = prometheus.NewGaugeVec(prometheus.GaugeOpts{
-		Name: "controller_runtime_max_concurrent_reconciles",
-		Help: "Maximum number of concurrent reconciles per controller",
-	}, []string{"controller"})
-	activeWorkers = prometheus.NewGaugeVec(prometheus.GaugeOpts{
-		Name: "controller_runtime_active_workers",
-		Help: "Number of currently used workers per controller",
-	}, []string{"controller"})
-)
```

Comment:
Can we make sure that these metrics are covered by operatorpkg? Since we're relying on operatorpkg to give us these metrics now, I'm worried about unexpectedly introducing metrics changes to users. thoughts?

Reply:
This was validated, yes. Operatorpkg isn't providing these metrics; it's controller-runtime that's doing it. We had to add these before because we were creating our own form of a "runnable" that wasn't using the standard controller framework with a `NewControllerManagedBy` call. Now that all of our controllers are making calls to this same function and are using the controller framework, there's no longer any need to statically define these metrics.

Reply:
Just so I understand, the set of metrics in operatorpkg will use the same controller-runtime metrics, and potentially add more of its own. Is that right?

Reply:
Just the controller-runtime ones. Doesn't add any more.